packages/consumption/test/modules/identityMetadata/IdentityMetadataController.test.ts

```diff
@@ -0,0 +1,172 @@
+        expect(queried!.value.toJSON()).toStrictEqual({ key: "value2" });
+    });
+
+    test("should upsert an identity metadata", async function () {
+        const query = { reference: CoreAddress.from("id1") };
+
+        const upserted = await consumptionController.identityMetadata.upsertIdentityMetadata({
+            ...query,
+            value: { key: "value2" }
+        });
+        expect(upserted.value.toJSON()).toStrictEqual({ key: "value2" });
+
+        const queried = await consumptionController.identityMetadata.getIdentityMetadata(CoreAddress.from("id1"));
+        expect(queried).toBeDefined();
+        expect(queried!.value.toJSON()).toStrictEqual({ key: "value2" });
+    });
+
+    test("should upsert an identity metadata with a key", async function () {
+        const query = {
+            reference: CoreAddress.from("id1"),
+            key: "key"
+        };
+
+        const upserted = await consumptionController.identityMetadata.upsertIdentityMetadata({
+            ...query,
+            value: { key: "value2" }
+        });
+        expect(upserted.value.toJSON()).toStrictEqual({ key: "value2" });
+
+        const queried = await consumptionController.identityMetadata.getIdentityMetadata(CoreAddress.from("id1"), "key");
+        expect(queried).toBeDefined();
+        expect(queried!.value.toJSON()).toStrictEqual({ key: "value2" });
+    });
+
+    test("should delete an identity metadata", async function () {
+        const identityMetadata = await consumptionController.identityMetadata.upsertIdentityMetadata({
+            reference: CoreAddress.from("id1"),
+            value: { key: "value" }
+        });
+
+        expect(await consumptionController.identityMetadata["identityMetadata"].count()).toBe(1);
+
+        await consumptionController.identityMetadata.deleteIdentityMetadata(identityMetadata);
+        expect(await consumptionController.identityMetadata["identityMetadata"].count()).toBe(0);
+    });
+
+    test("should delete an identity metadata with a key", async function () {
+        const identityMetadata = await consumptionController.identityMetadata.upsertIdentityMetadata({
+            reference: CoreAddress.from("id1"),
+            key: "key",
+            value: { key: "value" }
+        });
+
+        expect(await consumptionController.identityMetadata["identityMetadata"].count()).toBe(1);
+
+        await consumptionController.identityMetadata.deleteIdentityMetadata(identityMetadata);
+        expect(await consumptionController.identityMetadata["identityMetadata"].count()).toBe(0);
+    });
+
+    test("should delete all identity metadata exchanged with a peer", async function () {
+        await consumptionController.identityMetadata.upsertIdentityMetadata({ reference: CoreAddress.from("address1"), value: { key: "value" } });
+        await consumptionController.identityMetadata.upsertIdentityMetadata({ reference: CoreAddress.from("address1"), value: { key: "value" }, key: "key" });
```

Comment:
At this point at the latest, I feel the need not to use `{ key: "value" }` as an example value for `value` :D